packages/sitecore-jss-nextjs/src/middleware/personalize-middleware.ts

```diff
@@ -154,15 +170,25 @@ export class PersonalizeMiddleware extends MiddlewareBase {
       return response;
     }
 
-    const site = this.getSite(req, res);
+    const site = this.getSite(req, response);
+    const engageServer = this.initializeEngageServer(site, language);
+
+    // creates the browser ID cookie on the server side
+    // and includes the cookie in the response header
+    try {
+      await engageServer.handleCookie(req, response, timeout);
+    } catch (error) {
+      debug.personalize('skipped (browser id generation failed)');
+      console.log(error);
```

Comment:
If we are throwing the error we don't need this log here; it will get logged already in `getHandler`. Otherwise, we can `return response` here instead of throwing like we did previously. I like the throwing.